driver-core/src/main/com/mongodb/connection/SocketStreamFactory.java

```diff
@@ -63,16 +64,20 @@ public SocketStreamFactory(final SocketSettings settings, final SslSettings sslS
     @Override
     public Stream create(final ServerAddress serverAddress) {
         Stream stream;
-        if (socketFactory != null) {
-            stream = new SocketStream(serverAddress, settings, sslSettings, socketFactory, bufferProvider);
-        } else if (sslSettings.isEnabled()) {
-            stream = new SocketStream(serverAddress, settings, sslSettings, getSslContext().getSocketFactory(), bufferProvider);
-        } else if (System.getProperty("org.mongodb.useSocket", "false").equals("true")) {
-            stream = new SocketStream(serverAddress, settings, sslSettings, SocketFactory.getDefault(), bufferProvider);
+        if (serverAddress instanceof UnixServerAddress) {
+            if (sslSettings.isEnabled()) {
```

Comment:
Also throw if socketFactory != null

Reply:
The driver will create a SocketFactory if one doesn't exist - so can't throw.